nearcore/src/config_validate.rs

```diff
@@ -0,0 +1,146 @@
+use near_config_utils::{ValidationError, ValidationErrors};
+
+use crate::config::Config;
+
+/// Validate Config extracted from config.json.
+/// This function does not panic. It returns the error if any validation fails.
+pub fn validate_config(config: &Config) -> Result<(), ValidationError> {
+    let mut validation_errors = ValidationErrors::new();
+    let mut config_validator = ConfigValidator::new(config, &mut validation_errors);
+    println!("\nValidating Config, extracted from config.json...");
```

Comment:
Please avoid `println!()` as all output of neard normally goes to stderr, which is done via the `tracing` create.

```suggestion
    tracing::info!(target: "nearcore", "Validating Config, extracted from config.json...");
```

Reply:
is there a rule as to what we put for `target`? Searching in nearcore, seems like it does not always match the name of the library the code sits in...

Reply:
I'm putting `config` as target for all messages related to config validation

Reply:
We have some guidelines https://github.com/near/nearcore/blob/master/docs/practices/style.md#tracing
Avoid creating unnecessary targets, because filtering them in `RUST_LOG` will be too much work.